ddl/placement/rule.go

```diff
@@ -62,6 +62,23 @@ func NewRule(role PeerRoleType, replicas uint64, cnst Constraints) *Rule {
 	}
 }
 
+func getYamlMapFormatError(str string) error {
+	if strings.Contains(str, ":") {
+		findSeparator := false
+		for i := 0; i < len(str); i = i + 1 {
+			if findSeparator && str[i] != ' ' {
+				return ErrMappingWrongSeparator
+			}
+			findSeparator = false
+			if str[i] == ':' {
+				findSeparator = true
+			}
+		}
+		return nil
+	}
+	return ErrMappingNoColonFound
```

Comment:
I recommended to use a pre compiled regexp like `[^"':]+:\d` to match.

Reply:
I think you are right.

Reply:
Done